whisperv6/whisper.go

```diff
@@ -891,22 +901,36 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 
 				requestID = common.BytesToHash(payload[:common.HashLength])
 
-				if len(payload) >= common.HashLength*2 {
-					lastEnvelopeHash = common.BytesToHash(payload[common.HashLength : common.HashLength*2])
+				if isErrorPayload(payload) {
+					whisper.envelopeFeed.Send(EnvelopeEvent{
+						Hash:  requestID,
+						Event: EventMailServerRequestFailed,
+						Data:  string(payload[common.HashLength:]),
```

Comment:
Should we also get rid of the `ERROR=` prefix?

Reply:
oh, right... tomorrow! :)